plugins/tiddlywiki/markdown/EditorToolbar/file-import.tid

```diff
@@ -0,0 +1,51 @@
+title: $:/plugins/tiddlywiki/markdown/EditorToolbar/file-import
+tags: $:/tags/EditorTools
+condition: [<targetTiddler>type[text/x-markdown]] [<targetTiddler>type[text/markdown]]
+
+\procedure lingo-base() $:/language/Import/
+
+\procedure closePopupActions()
+<$action-deletetiddler $filter="[title<importState>] [title<importTitle>]"/>
+\end
+
+\procedure replacement-text-image() ![](<#$title$>)
+
+\procedure replacement-text-file() [](<#$title$>)
+
+\procedure replacement-regexp() [()<>\\]
+
+\procedure postImportActions()
+\whitespace trim
+<$list filter="[<importTitle>links[]]">
+<$let escaped-title={{{ [<currentTiddler>search-replace:g:regexp<replacement-regexp>,[\$&]] }}}>
+<$let insertText={{{ [get[type]prefix[image]then<replacement-text-image>else<replacement-text-file>search-replace[$title$],<escaped-title>] }}}>
+<$action-sendmessage
+	$message="tm-edit-text-operation"
+	$param="insert-text"
+	text=<<insertText>>
+/>
+</$let>
+</$let>
+</$list>
+<<closePopupActions>>
+\end
+
+\procedure buttons()
+\whitespace trim
+<$button class="tc-btn-invisible" actions=<<closePopupActions>> ><<lingo Listing/Cancel/Caption>></$button>
+&#32;
```

Comment:
We try to avoid HTML whitespace characters to define spacings between elements. 
You can use `class= "tc-btn-invisible tc-tiny-gap-right"` or `tc-small-gap-right` to define margins between elements. 

There are several `tc-*` [definitions in the vanilla/base CSS](https://github.com/pmario/TiddlyWiki5/blob/3fbaf85da7254c5008d55e907e0b42c165f9a373/themes/tiddlywiki/vanilla/base.tid#L3459)

Reply:
I think we can create a new PR to modify `$:/core/ui/EditorToolbar/file-import` (replace macros with procedures, replace HTML whitespace characters with utility classes) if this PR is merged.